### PR TITLE
Fluent step2

### PR DESCRIPTION
**Pull request by christianalfoni**
- The project is now configured with Typescript transpiling using webpack and ts-loader, optimized with **fork-ts-checker-webpack-plugin**
- The entry file is changed to the `index.tsx` file
- The **fluent** file is configured
- Main module and patron is refactored with state types
- Components related to running the Patron page is now updated to Typescript, using **fluent connect**

The way this can move forward is that we merge in changes as we go. We take one module and/or page at a time.

### Module change
1. Create a `types.ts` file
2. Add **State** type, using the previous `model.js` file as reference
3. Move the state object of the module file to a variable where **State** from types is used as type
4. Go to **fluent** file and compose in the module state

### Component change
1. Replace **inject** and **observer** from `mobx-react` with **connect** from the fluent file (`app/fluent`)
2. Connect the state and signals needed. Currently signals has an `any` type, as these types will come in step 3

That is basically it. The **Patron** page and module is a really good reference for the change that has been made :)

### CODE REVIEW

File: packages/codesandbox-api/src/dispatcher/index.ts [hunk 1]
```diff
@@ -15,22 +14,22 @@ export const isStandalone =
...
-  }
+    if (window.opener) {
+        window.opener.postMessage(newMessage, '*');
+    } else {
+        window.parent.postMessage(newMessage, '*');
+    }
 }
 
-export type Callback = (message: Object, source?: Window) => void;
+export type Callback = (message: Object, source?: any) => void;
```
Comment (CompuIves) [inline packages/codesandbox-api/src/dispatcher/index.ts]: Hmm, why is this one an `any`?

File: packages/app/src/app/components/CodeEditor/CodeMirror/elements.ts [hunk 2]
```diff
@@ -30,7 +35,7 @@ export const CodeContainer = styled.div`
     font-family: ${props =>
       fontFamilies(props.fontFamily, 'Menlo', 'Source Code Pro', 'monospace')};
     line-height: ${props => props.lineHeight};
-    background: ${theme.background2()};
+    background: ${props => props.theme.background2()};
```
Comment (CompuIves) [inline packages/app/src/app/components/CodeEditor/CodeMirror/elements.ts]: I think we need to use `theme` here directly since this Button can be used outside of the app context.

File: packages/app/src/app/components/CodeEditor/CodeMirror/index.tsx [hunk 3]
```diff
@@ -14,35 +15,53 @@ import 'codemirror/addon/lint/lint';
 
 import FuzzySearch from '../FuzzySearch';
 import { Container, CodeContainer } from './elements';
+import { Sandbox, Module, SandboxError } from 'app/store/modules/editor/types';
```
Comment (CompuIves) [inline packages/app/src/app/components/CodeEditor/CodeMirror/index.tsx]: I really like this!

File: packages/app/src/app/components/CodeEditor/ImageViewer/index.tsx [hunk 4]
```diff
@@ -1,24 +1,25 @@
...
 import { Container, Title, SubTitle, Image, MaxWidth } from './elements';
+import { Props as CodeEditorProps } from '../';
 
-import type { Props, Editor } from '../types';
+type Props = CodeEditorProps & {
+  dependencies: any;
+};
 
-export default class ImageViewer extends React.Component<Props>
-  implements Editor {
```
Comment (CompuIves) [inline packages/app/src/app/components/CodeEditor/ImageViewer/index.tsx]: I think we lose the autocompletion of functions of the editor of we remove this implements.

File: packages/app/src/app/components/CodeEditor/Monaco/elements.ts [hunk 5]
```diff
@@ -27,7 +26,7 @@ export const CodeContainer = styled.div`
       content: '';
       width: 50%;
       height: 50%;
-      background-color: ${theme.green()};
+      background-color: ${props => props.theme.green()};
```
Comment (CompuIves) [inline packages/app/src/app/components/CodeEditor/Monaco/elements.ts]: Same thing as button. but maybe this is not the case anymore. I will double check this!

File: packages/app/src/app/components/CodeEditor/Monaco/index.tsx [hunk 6]
```diff
@@ -151,13 +229,6 @@ class MonacoEditor extends React.Component<Props, State> implements Editor {
     this.lintWorker = null;
     this.typingsFetcherWorker = null;
     this.sizeProbeInterval = null;
-
-    this.resizeEditor = debounce(this.resizeEditor, 500);
```
Comment (CompuIves) [inline packages/app/src/app/components/CodeEditor/Monaco/index.tsx]: Why are these removed?

File: packages/app/src/app/components/CodeEditor/Monaco/index.tsx [hunk 7]
```diff
@@ -211,11 +282,11 @@ class MonacoEditor extends React.Component<Props, State> implements Editor {
 
     requestAnimationFrame(() => {
       this.setupWorkers();
-      editor.onDidChangeModelContent(e => {
+      editor.onDidChangeModelContent(({ changes }) => {
         const { isLive, sendTransforms } = this.props;
 
         if (isLive && sendTransforms && !this.receivingCode) {
-          this.sendChangeOperations(e);
+          this.addChangesOperation(changes);
```
Comment (CompuIves) [inline packages/app/src/app/components/CodeEditor/Monaco/index.tsx]: This should be `sendChangeOperations(e)` IIRC

File: packages/app/src/app/components/CodeEditor/Monaco/index.tsx [hunk 8]
```diff
@@ -239,18 +310,17 @@ class MonacoEditor extends React.Component<Props, State> implements Editor {
     this.addKeyCommands();
     import(/* webpackChunkName: 'monaco-emmet' */ './enable-emmet').then(
       enableEmmet => {
-        enableEmmet.default(editor, monaco, {});
+        enableEmmet.default(editor, monaco);
```
Comment (CompuIves) [inline packages/app/src/app/components/CodeEditor/Monaco/index.tsx]: I think this one should have the `, {}`

File: packages/app/src/app/components/CodeEditor/Monaco/index.tsx [hunk 9]
```diff
@@ -327,18 +398,6 @@ class MonacoEditor extends React.Component<Props, State> implements Editor {
     if (this.props.onInitialized) {
       this.disposeInitializer = this.props.onInitialized(this);
     }
-
-    // TODO remove this as soon as we solve the keybinding issues
-    editor.addCommand(
```
Comment (CompuIves) [inline packages/app/src/app/components/CodeEditor/Monaco/index.tsx]: This one shoukd stay.

File: packages/app/src/app/components/CodeEditor/Monaco/index.tsx [hunk 10]
```diff
@@ -430,85 +489,105 @@ class MonacoEditor extends React.Component<Props, State> implements Editor {
         // that the changes of code are not sent to live users. We need to reset
         // this state when we're doing changing modules
         this.props.onCodeReceived();
-        this.liveOperationCode = '';
```
Comment (CompuIves) [inline packages/app/src/app/components/CodeEditor/Monaco/index.tsx]: This should stay.

File: packages/app/src/app/components/CodeEditor/Monaco/index.tsx [hunk 11]
```diff
@@ -430,85 +489,105 @@ class MonacoEditor extends React.Component<Props, State> implements Editor {
...
+    changes.forEach(change => {
+      this.changes.changes.push(change);
+    });
+
+    this.sendChangeOperations();
   };
 
-  liveOperationCode = '';
-  sendChangeOperations = changeEvent => {
+  sendChangeOperations = (retry: boolean = false) => {
```
Comment (CompuIves) [inline packages/app/src/app/components/CodeEditor/Monaco/index.tsx]: We should have the other code here, so `sendChangeOperations = changeEvent` and `liveOperationCode = ''` etc.

File: packages/app/scripts/start.js [hunk 12]
```diff
@@ -95,21 +95,19 @@ function setupCompiler(port, protocol) {
     // We use stats.toJson({}, true) to make output more compact and readable:
     // https://github.com/facebookincubator/create-react-app/issues/401#issuecomment-238291901
     var json = stats.toJson({}, true);
-    var formattedErrors = json.errors.map(
-      message => 'Error in ' + formatMessage(message)
-    );
+    var formattedErrors = json.errors
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/scripts/start.js]: It is because errors or not only at the root of json. The children can also have errors. This is where I got stunned a bit trying to figure out some type errors I believe it was. It was located on the children error. So yeah. It is still the same, just reduces in the children errors as well :)

File: packages/app/src/app/components/CodeEditor/CodeMirror/elements.ts (hunk 2 above)
Comment (christianalfoni) [reply packages/app/src/app/components/CodeEditor/CodeMirror/elements.ts]: Ah, I see, okay, let me fix that :)

File: packages/app/src/app/components/CodeEditor/ImageViewer/index.tsx (hunk 4 above)
Comment (christianalfoni) [reply packages/app/src/app/components/CodeEditor/ImageViewer/index.tsx]: Ah, yeah, I thought it was unlikely to change again. But yeah, better to have it there. Actually ImageViewer does not implement anything from Editor, so get lint error there. But put it back on Codemirror and Monaco :)

File: packages/app/src/app/components/CodeEditor/Monaco/index.tsx (hunk 8 above)
Comment (christianalfoni) [reply packages/app/src/app/components/CodeEditor/Monaco/index.tsx]: There is no third argument looking at `enableEmmet` function :)

File: packages/app/src/app/components/CodeEditor/CodeMirror/index.tsx [hunk 13]
```diff
@@ -14,35 +15,53 @@ import 'codemirror/addon/lint/lint';
...
-  cm: typeof CodeMirror,
-  highlightedLines: Array<number>
-) => {
+const highlightLines = (cm: CodeMirror.Editor, highlightedLines: number[]) => {
   highlightedLines.forEach(line => {
     cm.addLineClass(+line - 1, 'background', 'cm-line-highlight');
   });
 };
 
-class CodemirrorEditor extends React.Component<Props, State> implements Editor {
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/components/CodeEditor/CodeMirror/index.tsx]: 👍

File: packages/app/src/app/components/ContextMenu/elements.ts [hunk 14]
```diff
@@ -1,12 +1,11 @@
...
-import theme from 'common/theme';
+import styled from 'app/styled-components';
 
 export const Container = styled.div`
   position: fixed;
 
   font-size: 0.875rem;
-  background-color: ${() => theme.background2.lighten(0.2)()};
-  color: ${() => theme.background2.lighten(3)()};
+  background-color: ${props => props.theme.background2.lighten(0.2)()};
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/components/ContextMenu/elements.ts]: I just changed it for now :)

File: packages/app/src/app/components/Preview/DevTools/Console/index.tsx [hunk 15]
```diff
@@ -10,19 +10,30 @@ import Input from './Input';
 
 import { Container, Messages } from './elements';
 
-export type IMessage = {
-  type: 'message' | 'command' | 'return',
-  logType: 'log' | 'warn' | 'info' | 'error',
-  arguments: any[],
+export type Message = {
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/components/Preview/DevTools/Console/index.tsx]: Yeah, I have not really thought about this. I see people use T and I convention, but also see those same people just use T in generic etc. I like to be really explicit. But yeah... actually showing type in the name using T and I is perfectly okay by me, though we should do it all over I think. Do not have the motivation to run through that now, but we can at least start doing it and fix it if we find wrong usage. I have fixed this one for now :)

File: packages/app/src/app/store/modules/live/computed.ts [hunk 16]
```diff
@@ -0,0 +1,41 @@
+import { State as RootState } from 'app/fluent';
+import { State } from './types';
+
+export function isEditor(state: State, rootState: RootState) {
+  return (
+    state.isLive &&
+    (state.roomInfo.mode === 'open' ||
+      state.roomInfo.ownerId === rootState.user.id ||
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/store/modules/live/computed.ts]: Ah, sorry, I see... my mistake. Let me fix this :) I did this:

File: packages/app/src/app/components/CodeEditor/Monaco/index.tsx [hunk 17]
```diff
@@ -308,15 +360,15 @@ class MonacoEditor extends React.Component<Props, State> implements Editor {
...
+          this.onSelectionChangedDebounce.cancel();
           onSelectionChanged({
             selection: data,
             moduleShortid: this.currentModule.shortid,
           });
         } else {
           // This is just on typing, we send a debounced selection update as a
           // safeguard to make sure we are in sync
-          this.onSelectionChangedDebounced({
+          this.onSelectionChanged({
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/components/CodeEditor/Monaco/index.tsx]: Oh, sorry. Yeah, I had some problems with the typing here and got a bit confused. Fixed now :)

File: packages/app/src/app/pages/common/Modals/ShareModal/index.tsx [hunk 18]
```diff
@@ -43,13 +66,14 @@ class ShareView extends React.Component {
...
     showQRCode: false,
+    message: '',
   };
 
   handleChange = e => this.setState({ message: e.target.value });
 
   handleSend = () => {
     if (this.state.message !== '') {
-      this.toggle();
+      // this.toggle();
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/common/Modals/ShareModal/index.tsx]: There is no method `toggle` on that component. Not sure how this piece of code could have worked?

File: packages/app/src/app/components/CodeEditor/Monaco/index.tsx (hunk 8 above)
Comment (CompuIves) [reply packages/app/src/app/components/CodeEditor/Monaco/index.tsx]: Ah lol, sorry! My bad 😄

File: packages/app/src/app/components/CodeEditor/CodeMirror/index.tsx (hunk 13 above)
[the remark this answers is not in the document]
Comment (CompuIves) [reply packages/app/src/app/components/CodeEditor/CodeMirror/index.tsx]: ❤️

File: packages/app/src/app/components/Preview/DevTools/Console/index.tsx [hunk 19]
```diff
@@ -112,18 +121,14 @@ class Console extends React.Component {
...
     if (nextProps.sandboxId !== this.props.sandboxId) {
       this.clearConsole();
     }
   }
 
   clearConsole = () => {
-    if (this.props.updateStatus) {
-      this.props.updateStatus('clear');
-    }
+    this.props.updateStatus('clear');
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/components/Preview/DevTools/Console/index.tsx]: Okay, fixed

File: packages/app/src/app/components/Preview/DevTools/Console/index.tsx [hunk 20]
```diff
@@ -95,10 +106,8 @@ class Console extends React.Component {
     return 'error';
   };
 
-  addMessage(message, args, type) {
-    if (this.props.updateStatus) {
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/components/Preview/DevTools/Console/index.tsx]: Okay, fixed

File: packages/app/src/app/components/Preview/DevTools/Tests/index.tsx [hunk 21]
```diff
@@ -284,52 +270,45 @@ class Tests extends React.Component<Props, State> {
...
-  } = {};
-  getStatus = (file: ?File): Status => {
+  getStatus = (file?: File): Status => {
     if (file == null) {
-      return 'idle';
+      return Status.IDLE;
     }
 
-    const lastFile = this._lastFiles[file.fileName];
+    const lastFile = this.lastFiles[file.fileName];
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/components/Preview/DevTools/Tests/index.tsx]: I just got linting errors on it previously, but not now with eslint, changed back :)

File: packages/app/src/app/components/Preview/DevTools/index.tsx [hunk 22]
```diff
@@ -41,59 +40,50 @@ const PANES = {
...
 };
 
 export default class DevTools extends React.PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
-
-    const hasView = props.view && props.view !== 'browser';
-
-    let currentPane = PANES[Object.keys(PANES)[0]].title;
-    if (hasView) {
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/components/Preview/DevTools/index.tsx]: Okay, just refactored this whole file from master

File: packages/app/src/app/components/Preview/DevTools/index.tsx [hunk 23]
```diff
@@ -196,13 +188,13 @@ export default class DevTools extends React.PureComponent<Props, State> {
     });
   };
 
-  handleTouchStart = (event: TouchEvent) => {
+  handleTouchStart = (event: any) => {
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/components/Preview/DevTools/index.tsx]: Okay, just refactored this whole file from master... this was a really annoying thing, so many different types of events. React, mouse, touch etc. had to ignore some stuff

File: packages/app/src/app/components/Preview/DevTools/index.tsx [hunk 24]
```diff
@@ -235,6 +227,9 @@ export default class DevTools extends React.PureComponent<Props, State> {
         e.stopPropagation();
         this.handleClick();
       } else {
+        // We do this to force a recalculation of the iframe height, this doesn't
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/components/Preview/DevTools/index.tsx]: Okay, just refactored this whole file from master

File: packages/app/src/app/components/Preview/DevTools/index.tsx [hunk 25]
```diff
@@ -388,7 +382,7 @@ export default class DevTools extends React.PureComponent<Props, State> {
             <Content
               key={title}
               hidden={hidden || title !== this.state.currentPane}
-              updateStatus={this.updateStatus(title)}
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/components/Preview/DevTools/index.tsx]: Okay, just refactored this whole file from master

File: packages/app/src/app/components/Preview/DevTools/Tests/index.tsx [hunk 26]
```diff
@@ -141,11 +149,11 @@ class Tests extends React.Component<Props, State> {
             f => this.getStatus(this.state.files[f]) === Status.PASS
           ).length;
 
-          if (failingTests > 0) {
+          if (this.props.updateStatus && failingTests > 0) {
```
Comment (CompuIves) [inline packages/app/src/app/components/Preview/DevTools/Tests/index.tsx]: This whole if statement block can be put in one `if (this.props.updateStatus)`

File: packages/app/src/app/pages/Sandbox/Editor/Content/Tabs/Tab/index.tsx [hunk 27]
```diff
@@ -69,27 +88,23 @@ class Tab extends React.Component {
...
+            show={tabCount > 1 && (active || hovering)}
           />
         )}
       </Container>
     );
   }
 }
 
-export default observer(Tab);
+export default Tab;
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Content/Tabs/Tab/index.tsx]: Could not see it observed anything, but another look now... it observes the module title :) Fixed

File: packages/app/src/app/pages/Sandbox/Editor/Content/index.tsx [hunk 28]
```diff
@@ -212,7 +204,7 @@ class EditorPreview extends React.Component<Props, State> {
         store.editor.pendingOperation &&
         store.editor.pendingOperation.map(x => x),
       () => {
-        if (store.editor.pendingOperation && store.live.isLive) {
+        if (store.editor.pendingOperation) {
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Content/index.tsx]: refactored whole file again

File: packages/app/src/app/pages/Sandbox/Editor/Content/index.tsx [hunk 29]
```diff
@@ -245,18 +238,12 @@ class EditorPreview extends React.Component<Props, State> {
     );
 
     const updateUserSelections = () => {
-      if (store.editor.pendingUserSelections) {
-        if (editor.updateUserSelections) {
-          if (store.live.isLive) {
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Content/index.tsx]: refactored whole file again

File: packages/app/src/app/pages/Sandbox/Editor/Content/index.tsx [hunk 30]
```diff
@@ -444,13 +432,7 @@ class EditorPreview extends React.Component<Props, State> {
                 this.devtools = component;
               }
             }}
-            setDragging={dragging => {
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Content/index.tsx]: refactored whole file again

File: packages/app/src/app/pages/Sandbox/Editor/Header/Logo/index.tsx [hunk 31]
```diff
@@ -4,11 +4,17 @@ import LogoIcon from 'common/components/Logo';
...
+type TLogo = React.SFC<{
+  title: string;
+}>;
+
+const Logo: TLogo = ({ title }) => (
   <Container id="logo" href="/">
     <div style={{ position: 'relative', display: 'flex' }}>
       <LogoIcon title="CodeSandbox" width={30} height={30} />
     </div>
-    <Title>Sandbox Editor</Title>
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Header/Logo/index.tsx]: Fixed

File: packages/app/src/app/pages/Sandbox/Editor/Header/index.tsx [hunk 32]
```diff
@@ -23,13 +24,13 @@ import Action from './Action';
...
 import { Container, Right, Left } from './elements';
 
-function Header({ store, signals }) {
+const Header: React.SFC<WithCerebral> = ({ store, signals }) => {
   const sandbox = store.editor.currentSandbox;
 
   return (
     <Container>
       <Left>
-        <Logo />
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Header/index.tsx]: fixed

File: packages/app/src/app/pages/Sandbox/Editor/Header/Action/index.tsx [hunk 33]
```diff
@@ -84,9 +89,7 @@ function ActionComponent({
         <ActionTooltip title={placeholder || tooltip}>
           <IconContainer>
             <Icon {...iconProps} />
-            {title !== undefined && (
-              <Title unresponsive={unresponsive}>{title}</Title>
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Header/Action/index.tsx]: Title has no `unresponsive` prop related to it:

```js
export const Title = styled.span`
  padding-left: 0.5rem;
`;
```

File: packages/app/src/app/pages/Sandbox/Editor/Navigation/index.tsx [hunk 34]
```diff
@@ -23,10 +21,11 @@ const IDS_TO_ICONS = {
   live: LiveIcon,
 };
 
-const Navigation = ({ store, signals }) => (
+const Navigation: React.SFC<WithCerebral> = ({ store, signals }) => (
   <Container>
-    {getWorkspaceItems(store)
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Navigation/index.tsx]: I changed out the logic. It is rather a `Computed` that evaluates the properties. Makes more sense :) Check at:

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/Files/index.tsx [hunk 35]
```diff
@@ -61,4 +53,4 @@ class Files extends React.Component {
   }
 }
 
-export default inject('signals', 'store')(observer(Files));
+export default Files;
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Workspace/Files/index.tsx]: Yeah, very likely, let me fix :)

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/Files/index.tsx [hunk 36]
```diff
@@ -50,8 +44,6 @@ class Files extends React.Component {
           innerRef={el => {
             this.directory = el;
           }}
-          title={sandbox.title || 'Project'}
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Workspace/Files/index.tsx]: Hm, nah, inside DirectoryEntry:

```js
    const title = root
      ? 'Project'
      : currentSandbox.directories.find(m => m.id === id).title;
```

Never points to a title prop :)

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/Files/index.tsx [hunk 37]
```diff
@@ -50,8 +44,6 @@ class Files extends React.Component {
...
-          openedModulesByUsers={openedModulesByUsers}
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Workspace/Files/index.tsx]: Hah... yeah, is like this in master as well 😂 Will keep it removed and can rather fix the bug during review :)

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/Files/index.tsx [hunk 38]
```diff
@@ -39,7 +34,6 @@ class Files extends React.Component {
         actions={
           <EditIcons
             hovering
-            forceShow={window.__isTouch}
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Workspace/Files/index.tsx]: Jup, is in now

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/index.tsx [hunk 39]
```diff
@@ -16,8 +15,6 @@ import NotOwnedSandboxInfo from './items/NotOwnedSandboxInfo';
 
 import ConnectionNotice from './ConnectionNotice';
 import Advertisement from './Advertisement';
-import WorkspaceItem from './WorkspaceItem';
-import Chat from './Chat';
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Workspace/index.tsx]: haha, yeah...chat is back :)

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/index.tsx [hunk 40]
```diff
@@ -1,9 +1,8 @@
 import * as React from 'react';
-import { inject, observer } from 'mobx-react';
+import { connect, WithCerebral } from 'app/fluent';
 
 import VERSION from 'common/version';
 
-import getWorkspaceItems from 'app/store/modules/workspace/items';
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Workspace/index.tsx]: Using computed here :)

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/index.tsx [hunk 41]
```diff
@@ -43,19 +40,13 @@ function Workspace({ store }) {
 
   const Component = sandbox.owned ? idToItem[currentItem] : NotOwnedSandboxInfo;
 
-  const item = getWorkspaceItems(store).find(i => i.id === currentItem);
+  const item = store.workspace.items.get().find(i => i.id === currentItem);
   return (
     <Container>
       {sandbox.owned && <ItemTitle>{item.name}</ItemTitle>}
-      <div style={{ flex: 1, overflowY: 'auto' }}>
+      <div style={{ flex: 1 }}>
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Workspace/index.tsx]: Refactored whole file, back now

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/index.tsx [hunk 42]
```diff
@@ -43,19 +40,13 @@ function Workspace({ store }) {
...
         <Component />
       </div>
-      {store.live.isLive &&
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Workspace/index.tsx]: Refactored whole file

File: packages/app/src/app/pages/Sandbox/Editor/Workspace/items/GitHub/index.tsx [hunk 43]
```diff
@@ -1,27 +1,23 @@
...
+  return store.user.integrations.github ? (
+    sandbox.originalGit ? ( // eslint-disable-line
       <Git />
     ) : (
       <CreateRepo />
     )
   ) : (
     <div>
-      <Description
-        margin={1}
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/Editor/Workspace/items/GitHub/index.tsx]: This is Description:

File: packages/app/src/app/pages/Sandbox/SearchDependencies/formatDownloads.js [hunk 44]
```diff
@@ -1,4 +1,4 @@
-export default function formatDownloads(downloads: number) {
+export default function formatDownloads(downloads) {
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/SearchDependencies/formatDownloads.js]: Not refactoring anything else than elements and components for now :)

File: packages/homepage/src/screens/home/NPMFeature/index.js [hunk 45]
```diff
@@ -109,74 +109,74 @@ const Feature = ({ right, title, Icon, iconSize, children }) => (
...
-          <Column flex={1}>
-            <ExtraSpacing>
-              <EmbedAnimation />
-            </ExtraSpacing>
-          </Column>
-        </FeatureRow>
-      </FeatureMargin>
-    </MaxWidth>
-  </Background>
+    <Background>
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/homepage/src/screens/home/NPMFeature/index.js]: I put back file from master

File: packages/homepage/src/pages/index.js [hunk 46]
```diff
@@ -8,6 +8,8 @@ import RecentPublications from '../screens/home/RecentPublications';
 import Patron from '../screens/home/Patron';
 import TitleAndMetaTags from '../components/TitleAndMetaTags';
 
+console.log(NPMFeature);
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/homepage/src/pages/index.js]: fixed

File: packages/common/templates/angular.js [hunk 47]
```diff
@@ -1,5 +1,3 @@
-// @flow
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/common/templates/angular.js]: Getting errors on them, due to new eslint config. but common is next in line for refactor, so will not be untyped for long :)

File: packages/app/src/app/store/sequences.js [hunk 48]
```diff
@@ -265,11 +266,10 @@ export const resetLive = [
   set(state`live.isLive`, false),
   set(state`live.error`, null),
   set(state`live.isLoading`, false),
-  set(state`live.isOwner`, false),
-  set(state`live.roomInfo`, undefined),
+  unset(state`live.roomInfo`),
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/store/sequences.js]: fixed

File: packages/app/src/app/store/sequences.js [hunk 49]
```diff
@@ -281,14 +281,13 @@ export const setSandbox = [
       },
     ],
   },
-  set(state`editor.currentId`, props`sandbox.id`),
+  set(state`editor.currentId`, props`id`),
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/store/sequences.js]: I think id is there as well, but put it back now. Will be nice with typing here later :)

File: packages/app/src/app/store/sequences.js [hunk 50]
```diff
@@ -107,7 +107,8 @@ export const forkSandbox = sequence('forkSandbox', [
   set(state`editor.isForkingSandbox`, true),
   actions.forkSandbox,
   actions.moveModuleContent,
-  set(state`editor.sandboxes.${props`sandbox.id`}`, props`sandbox`),
+  set(props`sandbox`, props`forkedSandbox`),
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/store/sequences.js]: Because with plain mobx we have to explicitly convert `npmDependencies` to a Dictionary, because it arrives as a plain object. With mobx-state-tree this seems to happen automatically with the "typing" model. But now with plain mobx we have to be explicit about this. Check `addSandbox` action... that is where it sets `npmdependencies` to dictionary

File: packages/app/src/app/pages/Sandbox/ZenModeIntroductionModal/index.tsx [hunk 51]
```diff
@@ -1,20 +1,20 @@
...
 import { Container, Heading, Explanation } from './elements';
 
-function ZenModeIntroduction({ signals }) {
+const ZenModeIntroduction: React.SFC<WithCerebral> = ({ signals }) => {
   return (
     <Container>
       <Heading>Zen Mode Explained</Heading>
       <Explanation>
         Zen Mode is perfect for giving instruction videos and presentations. You
-        can toggle the sidebar by double tapping <tt>shift</tt>. You can leave
```
[the remark this answers is not in the document]
Comment (christianalfoni) [reply packages/app/src/app/pages/Sandbox/ZenModeIntroductionModal/index.tsx]: Done
